java/driver/flight-sql/src/main/java/org/apache/arrow/adbc/driver/flightsql/InfoMetadataBuilder.java

```diff
@@ -145,7 +150,12 @@ VectorSchemaRoot build() throws AdbcException {
 
     root.setRowCount(dstIndex);
     VectorSchemaRoot result = root;
-    root = null;
+    try {
+      root = VectorSchemaRoot.create(StandardSchemas.GET_INFO_SCHEMA, allocator);
```

Comment:
Why does an empty root get recreated here?

FYI I'm reworking some of this code for the getObjects() impl. Planning to make this class an Iterator<VectorSchemaRoots> and have RootArrowReader consume the iterator to avoid front-loading the client when there are lots of metadata objects.